### PR TITLE
fix : requesting invoice with any amount and cleans code

### PR DESCRIPTION
**Pull request by Harshit933**
Removes the numberpad and button and adds the feature of requesting payments with having 'any' amount.

Potential fix : #134

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
Just come git comments

>Removes the numberpad and button and adds the feature of requesting payments with having 'any' amount.

Usually these are two commits

One for remove the button and the numberpad, and the other one for fixing any invoice

File: lib/views/pay/numberpad_view.dart [hunk 1]
```diff
@@ -93,143 +93,49 @@ class _NumberPadState extends State<NumberPad> {
...
+        child: NumberPadBody(
+          size: MediaQuery.of(context).size,
+          display: display,
+          onNumberButtonTap: (value, icon) {
+            if (icon != null) {
+              if (display.length == 1) {
+                display = '0';
+              } else {
+                display = display.substring(0, display.length - 1);
+              }
```
Comment (vincenzopalazzo) [inline lib/views/pay/numberpad_view.dart:112]: Why do not put the logic that we had as default inside the trash library? And if the user wants can override it by providing a `onNumberButtonTap` callback?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
I add a new consideration inside the review, thanks to do this semplification

File: lib/views/request/request_view.dart [hunk 2]
```diff
@@ -59,95 +77,66 @@ class _RequestViewState extends State<RequestView> {
...
+              setState(() {
+                display = int.parse(display + value).toString();
+              });
+            },
+            onPayButtonPress: () async {
+              if (display == '0') {
+                PopUp.showPopUp(
+                    context, 'Invalid amount', 'Enter an amount > 0', true);
+                return;
+              }
```
Comment (vincenzopalazzo) [inline lib/views/request/request_view.dart:118]: This PR looks untested, the check is still there

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Please test the PR before open it and request a review, your description is laying you are not removing the check to generate an invoice of any amount

File: lib/views/request/request_view.dart (hunk 2 above)
Comment (Harshit933) [reply lib/views/request/request_view.dart:118]: No no, actually I added a string at the last of the `request_view`. By clicking on it, you can create an invoice with any amount.

Comment (vincenzopalazzo) [reply lib/views/request/request_view.dart:118]: Oh no hahah I found it. Why not click on the request button and choose if you want a bitcoin or lightning address?

I think this is much better than finding a tiny label under the button. The user should not know if it is creating an invoice with any amount.

**Review by vincenzopalazzo — APPROVED** [review]
LGMT thanks

**Pull request merged**